deepwell/src/models/sea_orm_active_enums.rs

```diff
@@ -60,8 +64,12 @@ pub enum PageRevisionType {
     Move,
     #[sea_orm(string_value = "regular")]
     Regular,
+    #[sea_orm(string_value = "rollback")]
+    Rollback,
     #[sea_orm(string_value = "undelete")]
     Undelete,
+    #[sea_orm(string_value = "undo")]
+    Undo,
```

Comment:
What does `PageRevisionType::Undo` mean? Is it like the opposite of a rollback?

Reply:
Undos are a [planned operation](https://scuttle.atlassian.net/browse/WJ-990). I am avoiding the term "revert" because it means two different things in git vs wikidot, and instead have two unambiguous terms:

* Rollback (what wikidot calls "revert", what git calls "reset") is where you set the current state of a page or file to some revision in the past.
* Undo (what git calls "revert") is where it applies the opposite of the given revision. Since this involves applying a diff, it can fail.

An example of the difference  between the two. Imagine we have a page with this history:

<details>
<summary>Revision 10</summary>

```md
**Item #:** SCP-XX

**Object Class:** Euclid

**Special Containment Procedures:** SCP-XX is to be kept in a cardboard box.
```

</details>

<details>
<summary>Revision 11</summary>

```md
**Item #:** SCP-XX

**Object Class:** HAHA EPIC TROLL

**Special Containment Procedures:** SCP-XX is to be kept in a cardboard box.
```

</details>

</details>

<details>
<summary>Revision 12</summary>

```md
**Item #:** SCP-XX

**Object Class:** HAHA EPIC TROLL

**Special Containment Procedures:** SCP-XX is to be kept in a cardboard or paper box.
```

</details>

So here, revision 11 is vandalism and we'd like to remove it. But the page has been edited since, so that means we'd lose any subsequent (legitimate) changes with a simple rollback. But using undo, similar to `git revert`, we can remove just the specific change from 10 -> 11 and leave the 12 revision in-place.

So basically it would produce the 11..10 diff (not that this is the _inverse_ of the 10..11 diff that was applied in revision 11):
```diff
3c3
< **Object Class:** HAHA EPIC TROLL
---
> **Object Class:** Euclid
```

And then we could attempt to apply it.

(If there are conflicts, then like `git revert`, we should error out rather than attempting to ram our way through. Also, we should have some way of allowing a user to verify that the undo change is what they want before doing it.)